pypicloud/cache/dynamo.py

```diff
@@ -144,29 +255,33 @@ def summary(self):
                            key=lambda s: s.name)
         return [s.__json__() for s in summaries]
 
-    def clear(self, package):
-        summary = self.engine.get(PackageSummary, name=package.name)
-        if summary is not None and \
-            (summary.unstable == package.version or
-             summary.stable == package.version or
-             summary.last_modified == package.last_modified):
+    @retry(tries=3, exceptions=(ConditionalCheckFailedException,))
+    def _rebuild_summary_for_package_named(self, name):
+        """Rebuild the summary from the cached packages."""
+
+        pkgs = self.all(name)
```

Comment:
Hmm. Dynamo doesn't like consistent queries on GSIs. Still making the summary get consistent (since we expect this to clash).